src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancyProviders/AssetOccupancyProviderBase.cs

```diff
@@ -64,6 +64,62 @@ protected AssetOccupancyProviderBase(IServiceProvider serviceProvider)
     {
         var periodScheme = await GetEffectivePeriodSchemeAsync(targetDate, asset, categoryOfAsset);
 
+        return await InternalGetPeriodsAsync(asset, categoryOfAsset, periodScheme, targetDate, currentDateTime);
+    }
+
+    [UnitOfWork]
+    public virtual async Task<List<PeriodOccupancyModel>> GetPeriodsAsync(AssetCategory category,
+        Guid? periodSchemeId,
+        DateTime targetDate,
+        DateTime? currentDateTime = default)
+    {
+        var assets =
+            await AssetRepository.GetListAsync(x => x.AssetCategoryId == category.Id && !x.Disabled);
+
+        var effectivePeriodScheme = await GetEffectivePeriodSchemeAsync(periodSchemeId, category);
+
+        var models = effectivePeriodScheme.Periods
+            .Select(x =>
+                new PeriodOccupancyModel(targetDate,
+                    x.StartingTime,
+                    x.GetEndingTime(),
+                    effectivePeriodScheme.Id,
+                    x.Id,
+                    0, 0))
+            .ToList();
+
+        foreach (var asset in assets)
+        {
+            var assetPeriodScheme = await GetEffectivePeriodSchemeAsync(targetDate, asset, category);
+            if (effectivePeriodScheme.Id != assetPeriodScheme.Id)
+            {
+                continue;
+            }
+
+            var assetPeriodOccupancyModels =
+                await InternalGetPeriodsAsync(asset, category, assetPeriodScheme, targetDate, currentDateTime);
+
+            var assetPeriodOccupancyModelDictionary = assetPeriodOccupancyModels.ToDictionary(x => x.PeriodId);
+            foreach (var periodOccupancyModel in models)
+            {
+                var assetPeriodOccupancyModel = assetPeriodOccupancyModelDictionary[periodOccupancyModel.PeriodId];
+                if (periodOccupancyModel.AvailableVolume < assetPeriodOccupancyModel.AvailableVolume)
```

Comment:
Not everyone likes equalitarianism.

Refer to https://github.com/EasyAbp/BookingService/pull/31/files#r917280758, and don't forget to disorder the `assets`.

Setting all the `Priority` to `0` means "random select." If someone wants equalitarianism or more rules, he can implement `IAssetInCategorySelector` himself.